lib/fw.c

```diff
@@ -1075,7 +1075,7 @@ static int switchtec_fw_part_info(struct switchtec_dev *dev, int nr_info,
 	if (info == NULL || nr_info == 0)
 		return -EINVAL;
 
-	if (dev->gen == SWITCHTEC_GEN4) {
+	if (dev->gen != SWITCHTEC_GEN3) {
```

Comment:
Maybe we can use the style '> SWITCHTEC_GEN3', which implies that the change happened from GEN4. And when we have something new in GEN5 we can use the same style. Otherwise looks good to me. Thanks.

Reply:
Updated.